### PR TITLE
DS-4034 - The "first few letters" search doesn't work with diacritics

### LINKED ISSUE #7381: [DS-4034] The "first few letters" search doesn't work with diacritics
<i>Imported from JIRA <a href="https://jira.lyrasis.org/browse/DS-4034">[DS-4034]</a> created by adan.roman</i>
<p>It can be tested at <a href="http://demo.dspace.org/xmlui/browse?type=author" class="external-link" target="_blank" rel="nofollow noopener">http://demo.dspace.org/xmlui/browse?type=author</a> searching an author with diacritic "Azcárate" for example. It return no results.</p>

### PR DESCRIPTION
**Pull request by toniprieto**
https://jira.duraspace.org/browse/DS-4034
Fixes #7381

This pull request:

1) Changes the default normalization of the indexed text used for sorting and filtering the author and subject entries to also strip the diacritics.

The current default normalization only decompose the diacritics and doesn't work with the filter by searched value done by the "first few letters" search since the 6.3 version.

2) Applies the same normalization to the user-input text.

A re-index is needed for testing it.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/sort/OrderFormatAuthor.java [hunk 1]
```diff
@@ -21,6 +22,7 @@
 {
 	{
 		filters = new TextFilter[] { new DecomposeDiactritics(),
+				                     new StripDiacritics(),
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/sort/OrderFormatAuthor.java]: I could be wrong, but do we need *both* `DecomposeDiacritics()` and `StripDiacritics()` together?  Based on their descriptions, it seems like we should be using one or the other...but, maybe I'm misunderstanding this change?

File: dspace-api/src/main/java/org/dspace/sort/OrderFormatText.java [hunk 2]
```diff
@@ -21,6 +22,7 @@
 {
 	{
 		filters = new TextFilter[] { new DecomposeDiactritics(),
+			                         new StripDiacritics(),
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/sort/OrderFormatText.java]: Again, not sure if we should have ` DecomposeDiactritics()` and `StripDiacritics()` together.

File: dspace-api/src/main/java/org/dspace/sort/OrderFormatAuthor.java (hunk 1 above)
Comment (toniprieto) [reply dspace-api/src/main/java/org/dspace/sort/OrderFormatAuthor.java]: That's what I thought, too. But I can confirm that both are needed. StripDiacritics() needs decomposed diacritics to properly remove them.

This is explained in this old JIRA comment https://jira.duraspace.org/browse/DS-916?focusedCommentId=20559&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20559

Comment (tdonohue) [reply dspace-api/src/main/java/org/dspace/sort/OrderFormatAuthor.java]: @toniprieto : Thanks for that extra info! I wasn't aware of that.  This looks good then & I'll resolve both my comments.

File: dspace-api/src/main/java/org/dspace/sort/OrderFormatText.java (hunk 2 above)
Comment (tdonohue) [reply dspace-api/src/main/java/org/dspace/sort/OrderFormatText.java]: Resolved via discussion above.

**Review by tdonohue — APPROVED** [review]
👍 The code looks good to me. However, I have not had a chance to test this, so I'd appreciate someone else verifying via a test prior to merging.

**Review by J4bbi — APPROVED** [review]
@toniprieto , thanks for this pr

this is confirmed by me to work after testing

**Comment by tdonohue** [comment]
Ported to main in #2698

**Pull request merged**